compiler/rustc_type_ir/src/predicate.rs

```diff
@@ -196,6 +196,18 @@ impl fmt::Display for ImplPolarity {
     }
 }
 
+impl ImplPolarity {
+    /// Typically used to decorate the impl trait message by adding the corresponding
+    /// polarity marker before the trait name.
```

Comment:
```suggestion
```

I'm not super convinced that a doc comment adds any value. However if you'd like to keep it, I would shorten it to

```suggestion
    /// The polarity marker in front of the impl trait ref if applicable.
```